File: src/test/java/net/haesleinhuepf/clij/ops/examples/Minimum.java

package net.haesleinhuepf.clij.ops.examples;

import java.io.IOException;

import org.junit.Test;

import net.haesleinhuepf.clij.ops.CLIJ_close.CLIJ_close;
import net.haesleinhuepf.clij.ops.CLIJ_minimumBox.CLIJ_minimumBox;
import net.haesleinhuepf.clij.ops.CLIJ_pull.CLIJ_pull;
import net.haesleinhuepf.clij.ops.CLIJ_push.CLIJ_push;
import net.imagej.ImageJ;

public class Minimum {

	@Test
	public void run() throws IOException {
		ImageJ ij = new ImageJ();
		Object input = ij.io().open(getClass().getResource("/samples/t1-head.tif")
			.getPath());
		ij.ui().show("input", input);
		// push image to GPU
		Object inputGPU = ij.op().run(CLIJ_push.class, input);
		// minimum
		Object minimum = ij.op().run(CLIJ_minimumBox.class, inputGPU, 3, 3, 3);
		// show result
		Object result = ij.op().run(CLIJ_pull.class, minimum);
		ij.ui().show("minimum", result);

		// cleanup
		ij.op().run(CLIJ_close.class, inputGPU);
		ij.op().run(CLIJ_close.class, minimum);
		ij.op().run(CLIJ_close.class);

	}

	public static void main(String... args) throws IOException {
		Minimum task = new Minimum();
		task.run();
	}
}

File: src/test/resources/jython/minimum.py
# CLIJ example ImageJ Ops jython: minimum.py
#
# This macro shows how apply a minimum filter in the GPU.
#
# Author: Robert Haase, rhaase@mpi-cbg.de
# Author: Deborah Schmidt, frauzufall@mpi-cbg.de
# May 2019
# ---------------------------------------------

#@ IOService io
#@ UIService ui
#@ OpService ops

import os
import inspect
def getResource(file):
    return os.path.dirname(os.path.abspath(inspect.getsourcefile(lambda:0))) + "/" + file


input = io.open(getResource("../samples/t1-head.tif"))
ui.show("input", input)
# push image to GPU
inputGPU = ops.run("CLIJ_push", input)

minimum = ops.run("CLIJ_minimumBox", inputGPU, 3, 3, 3);
# show result
result = ops.run("CLIJ_pull", minimum)
ui.show("minimum", result)

#cleanup
ops.run("CLIJ_close", inputGPU)
ops.run("CLIJ_close", minimum)

